datumaro/plugins/coco_format/extractor.py

```diff
@@ -124,6 +161,55 @@ def _load_items(self, loader):
 
         return items
 
+    def _load_panoptic_items(self, config, panoptic_images):
+        import numpy as np
+        import PIL.Image
+        from collections import defaultdict
+        items = OrderedDict()
+
+        def rgb2id(color):
+            if isinstance(color, np.ndarray) and len(color.shape) == 3:
+                if color.dtype == np.uint8:
+                    color = color.astype(np.int32)
+                return color[:, :, 0] + 256 * color[:, :, 1] + 256 * 256 * color[:, :, 2]
+            return int(color[0] + 256 * color[1] + 256 * 256 * color[2])
+
+        imgs_info = {}
+        for img in config['images']:
+            imgs_info[img['id']] = img
+
+        for ann in config['annotations']:
+            img_id = ann['image_id']
+            image_path = osp.join(self._images_dir, imgs_info[img_id]['file_name'])
+            image_size = (imgs_info[img_id].get('height'), imgs_info[img_id].get('width'))
+            image = Image(path=image_path, size=image_size)
+            anns = list()
+
+            panoptic_image_path = osp.join(panoptic_images, ann['file_name'])
+            panoptic_format = np.array(PIL.Image.open(panoptic_image_path), dtype=np.uint32)
+            pan = rgb2id(panoptic_format)
+            rle_per_category = defaultdict(list)
+            for segm_info in ann['segments_info']:
+                cat_id = segm_info['category_id']
+                segm_id = segm_info['id']
+                mask = pan == segm_id
+                rle = mask_utils.encode(np.asfortranarray(mask.astype('uint8')))
+                rle['counts'] = rle['counts'].decode('utf8')
+                rle_per_category[cat_id].append(rle)
+
+            for cat_id, rle_list in rle_per_category.items():
+                rle = mask_utils.merge(rle_list)
```

Comment:
In Datumaro we provide a mask per instance, they need not to be merged. Mask `id` would be instance id. A bbox, if provided, should be an annotation of type `Bbox` with `group` = `id` of the mask (but having a separate bbox and area attributes seem useless, unless they have special meaning, because there are corresponding methods in the Mask class).

Reply:
I wanted to make real segmentation masks (as it's here, merged for different instances of one class) and panoptic masks (will be done later, it's important for export task)

Reply:
What is the use for merged masks of one class? Fully merged masks look attractive, but again, they are only useful for training, and they are interchangeable (by merging) with instance masks. In Datumaro, we decided to use instance masks, unless there is a strong need for compiled masks. And compiled masks can be obtained with `datumaro.components.extractor.CompiledMask`.